### PR TITLE
avoid platform matched /mswin|mingw|windows/ for Chef::Log::Syslog

### PR DESCRIPTION
**Pull request by sawanoboly**
Don't load `log/syslog` when platform is windows.

### CODE REVIEW

File: spec/unit/log_spec.rb [hunk 1]
```diff
@@ -23,21 +23,23 @@
 describe Chef::Log do
 end
 
-describe Chef::Log::Syslog do
-  let(:logger) { Chef::Log::Syslog.new }
+unless Chef::Platform.windows?
```
Comment (jdmundrawala) [inline spec/unit/log_spec.rb]: I think this needs to be rethought as this is a unit test

Comment (sawanoboly) [inline spec/unit/log_spec.rb]: Hmm, shall I move this to functional test ? @jdmundrawala

**Comment by sawanoboly** [comment]
@jdmundrawala move syslog specs from unit to functional and then update for call Chef::Log methods.

File: spec/unit/log_spec.rb (hunk 1 above)
Comment (lamont-granquist) [inline spec/unit/log_spec.rb]: In this case you can't use the syslog-logger gem on windows at all which makes it difficult to write a unit test that works on windows (not without getting crazy and stubbing require to return true and other horrible things).

**Comment by lamont-granquist** [comment]
@sawanoboly i'd say it was fine before as a unit test.

**Comment by sawanoboly** [comment]
@lamont-granquist So I'll just move spec/functional/log/syslog_spec.rb
 to spec/unit/log/syslog_spec.rb of 169e4de.

Would that be OK?

**Comment by lamont-granquist** [comment]
yep that works.

**Comment by sawanoboly** [comment]
Whoops! I have merged master with habit. I'll create new PR.

**Comment by sawanoboly** [comment]
@lamont-granquist Sorry, please refer #3322 . I'll close it.

**Pull request closed without merging**